devicetypes/smartthings/fidure-thermostat.src/fidure-thermostat.groovy

```diff
@@ -377,15 +379,13 @@ def updateSetpoint(attrib, val)
 
 	def value = '--';
 
+	if (("heat"  == mode && heat != null) ||
+    	("auto" == mode && runningMode == "heat" && heat != null))
+        	value = (attrib == "heatingSetpoint")? val : heat;
```

Comment:
doing this, makes the code compact, but makes it difficult to maintain. you are trying to compact too many ifs in two lines.